### PR TITLE
Fix token unlock bug for auctions

### PR DESCRIPTION
**Pull request by JasonTulp**
Currently when an auction is closed with no bids, the tokens are not unlocked and remain in a locked state. This PR addresses this

### CODE REVIEW

File: pallet/marketplace/src/impls.rs [hunk 1]
```diff
@@ -446,6 +446,7 @@ impl<T: Config> Pallet<T> {
 				},
 				Listing::Auction(listing) => {
 					Self::remove_listing(listing_outer, listing_id);
+					let _ = listing.tokens.unlock_tokens(&listing.seller);
```
Comment (zees-dev) [inline pallet/marketplace/src/impls.rs:449]: hmm `process_auction_closure` also does an unlock too.
i.e. `process_auction_closure` -> `process_payment_and_transfer` -> `tokens.unlock_and_transfer`

This implies that you may be doing redundant storage access (double removal).

A better solution maybe to only unlock the tokens in the `process_auction_closure` function within in this block:
```rust
let Some((winner, hammer_price)) = winning_bid else {
  // unlock here instead
  // ...
  return
}
```

**Review by zees-dev — COMMENTED** [review]
Everything else LGTM - just a small query WRT storage accesses.
You will also need to re-benchmark for any updated extrinsics.

File: pallet/marketplace/src/impls.rs (hunk 1 above)
Comment (leo-plugdefi) [reply pallet/marketplace/src/impls.rs:449]: agree here, as unlock should be part of the process_auction_closure

Comment (JasonTulp) [reply pallet/marketplace/src/impls.rs]: Good call, overlooked this. Addressed here: https://github.com/futureversecom/trn-seed/pull/826/commits/02a7b217a6f99b3f0735f50ad6491919cade69ff

**Review by zees-dev — APPROVED** [review]
LGTM 👍

**Review by leo-plugdefi — APPROVED** [review]

**Review by leo-plugdefi — COMMENTED** [review]
Looks like there is a conflict on migration files, and it seems like more folks are running into this snag right now. Hopefully, we can push it to the mainnet soon.

**Pull request merged**